bdb/queuedb.c

```diff
@@ -335,25 +555,45 @@ int bdb_queuedb_stats(bdb_state_type *bdb_state,
 
     dbt_key.flags = dbt_data.flags = DB_DBT_REALLOC;
 
-    rc = bdb_state->dbp_data[0][0]->cursor(bdb_state->dbp_data[0][0], NULL,
-                                           &dbcp, 0);
+    DB *db1 = BDB_QUEUEDB_GET_DBP_ZERO(bdb_state);
+    rc = db1->cursor(db1, NULL, &dbcp1, 0);
 
     if (rc != 0) {
         *bdberr = BDBERR_MISC;
-        return -1;
+        goto done;
     }
 
-    rc = bdb_cget_unpack(bdb_state, dbcp, &dbt_key, &dbt_data, &ver, DB_FIRST);
+    DB *db2 = BDB_QUEUEDB_GET_DBP_ONE(bdb_state);
+    if (db2 != NULL) {
+        rc = db2->cursor(db2, NULL, &dbcp2, 0);
+
+        if (rc != 0) {
+            *bdberr = BDBERR_MISC;
+            goto done;
+        }
+    } else {
+        dbcp2 = dbcp1; /* no second file, use same cursor */
+    }
+
+    rc = bdb_cget_unpack(bdb_state, dbcp1, &dbt_key, &dbt_data, &ver, DB_FIRST);
 
     if (rc) {
+        int did_dbcp2_first = 0;
+chk_db_first_rc:
         if (rc == DB_LOCK_DEADLOCK) {
             *bdberr = BDBERR_DEADLOCK;
             rc = -1;
             goto done;
         } else if (rc == DB_NOTFOUND) {
-            /* EOF */
-            rc = 0;
-            goto done;
+            if (!did_dbcp2_first && (dbcp2 != dbcp1)) {
+                rc = bdb_cget_unpack(bdb_state, dbcp2, &dbt_key, &dbt_data, &ver,
+                                     DB_FIRST);
+                did_dbcp2_first = 1; goto chk_db_first_rc;
```

Comment:
This is wrong- chk_db_first_rc goto target should be repositioned to line 579 .. 

>> this will error out unintentionally if file0 returns NOTFOUND, but file1 returns 0